homeassistant/components/zimi/light.py

```diff
@@ -0,0 +1,137 @@
+
+    entities = [ZimiLight(device, api) for device in api.lights]
+
+    async_add_entities(entities)
+
+
+class ZimiLight(LightEntity):
+    """Representation of a Zimi Light."""
+
+    def __init__(self, light: ControlPointDevice, api: ControlPoint) -> None:
+        """Initialize an ZimiLight."""
+
+        self._attr_unique_id = light.identifier
+        self._attr_should_poll = False
+        self._light = light
+        self._light.subscribe(self)
+        self._state = False
+        self._brightness = None
+        if self._light.type == "dimmer":
+            self._attr_color_mode = ColorMode.BRIGHTNESS
+            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+        else:
+            self._attr_color_mode = ColorMode.ONOFF
+            self._attr_supported_color_modes = {ColorMode.ONOFF}
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, light.identifier)},
+            name=self._light.name,
+            suggested_area=self._light.room,
+            via_device=(DOMAIN, api.mac),
+        )
+        self.update()
+        _LOGGER.debug("Initialising %s in %s", self.name, self._light.room)
+
+    @property
+    def name(self) -> str:
+        """Return the display name of this light."""
+        return self._light.name.strip()
+
+    @property
+    def available(self) -> bool:
+        """Return True if Home Assistant is able to read the state and control the underlying device."""
+        return self._light.is_connected
+
+    @property
+    def brightness(self) -> int | None:
+        """Return the brightness of the light.
+
+        This method is optional. Removing it indicates to Home Assistant
+        that brightness is not supported for this light.
+        """
+        return self._brightness
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if light is on."""
+        return self._state
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Instruct the light to turn on.
+
+        You can skip the brightness part if your light does not support
+        brightness control.
+        """
+
+        _LOGGER.debug(
+            "Sending turn_on(brightness=%d) for %s",
+            kwargs.get(ATTR_BRIGHTNESS, 255) * 100 / 255,
+            self.name,
+        )
+
+        if self._light.type == "dimmer":
+            await self._light.set_brightness(
+                kwargs.get(ATTR_BRIGHTNESS, 255) * 100 / 255
+            )
+        else:
+            await self._light.turn_on()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Instruct the light to turn off."""
+
+        _LOGGER.debug("Sending turn_off() for %s", self.name)
+
+        await self._light.turn_off()
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Cleanup ZimiLight with removal of notification prior to removal."""
+        self._light.unsubscribe(self)
+        return await super().async_will_remove_from_hass()
```

Comment:
```suggestion
        await super().async_will_remove_from_hass()
        self._light.unsubscribe(self)
```
to ensure super is always called